# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ProfilerHelpers.h
# Commit: e65d7e9dea10
# Full Hash: e65d7e9dea10c417588224abb452aa2d181e8e0c
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-01-13 15:45:24
# Regressor Bug: 1600906
# File Overlap Count: 1
# Description:
#   Bug 1600906 - Convert IDBCursor and BackgroundCursorChild to templates to increase type safety and reduce state. r=dom-workers-and-storage-reviewers,ytausky
#   
#   This also simplifies delegating calls that are dependent on the cursor type.
#   
#   Also reduce dependency on IDBCursor.h by moving enums and type traits to IDBCursorType.h
# ==============================================================================

diff -r 422de232c5ce -r e65d7e9dea10 dom/indexedDB/ProfilerHelpers.h
--- a/dom/indexedDB/ProfilerHelpers.h	Fri Jan 10 15:59:51 2020 +0000
+++ b/dom/indexedDB/ProfilerHelpers.h	Fri Jan 10 15:23:52 2020 +0000
@@ -12,7 +12,7 @@
 
 #include "BackgroundChildImpl.h"
 #include "GeckoProfiler.h"
-#include "IDBCursor.h"
+#include "IDBCursorType.h"
 #include "IDBDatabase.h"
 #include "IDBIndex.h"
 #include "IDBKeyRange.h"
@@ -189,18 +189,18 @@
     }
   }
 
-  explicit LoggingString(const IDBCursor::Direction aDirection) {
+  explicit LoggingString(const IDBCursorDirection aDirection) {
     switch (aDirection) {
-      case IDBCursor::Direction::Next:
+      case IDBCursorDirection::Next:
         AssignLiteral("\"next\"");
         break;
-      case IDBCursor::Direction::NextUnique:
+      case IDBCursorDirection::Nextunique:
         AssignLiteral("\"nextunique\"");
         break;
-      case IDBCursor::Direction::Prev:
+      case IDBCursorDirection::Prev:
         AssignLiteral("\"prev\"");
         break;
-      case IDBCursor::Direction::PrevUnique:
+      case IDBCursorDirection::Prevunique:
         AssignLiteral("\"prevunique\"");
         break;
       default: